### ISSUE TITLE
Unscheduled reboots during late night

### ISSUE THREAD

**Issue by johnwc** [weaveworks/kured#563]
I had a notification sent the night before last, that kured reboot one of our nodes in the middle of the night. Then last night, we got two more notifications about two more different nodes getting rebooted. We did not schedule the reboot, and no file existed. Is kured set to automatically reboot a node every x days/hours/weeks? The only two parameters we used during install was `period` and the notification url.

**Comment by johnwc** [comment]
Tell me how to get the logs. When viewing a pod log it only gives me few lines.

**Comment by ckotzbauer** [comment]
`kubectl logs -n <kured-namespace> <kured-pod-of-rebooted-node>` This shows the logs of the current running pod. Are the log-timestamps before or after the reboot?
I assume, the notification you got was from kured and not from a monitoring system?

**Comment by johnwc** [comment]
Sorry it took so long to respond, was pulled away on another project. It just happened again tonight.

```log
...
time="2022-06-09T06:23:38Z" level=info msg="Reboot not required"
time="2022-06-09T06:38:38Z" level=info msg="Reboot not required"
time="2022-06-09T06:53:38Z" level=info msg="Reboot not required"
time="2022-06-09T07:08:38Z" level=info msg="Reboot required"
time="2022-06-09T07:08:39Z" level=info msg="Acquired reboot lock"
time="2022-06-09T07:08:39Z" level=info msg="Draining node rnt1-worker2-69c6dcac-qmtzr"
WARNING: ignoring DaemonSet-managed Pods: calico-system/calico-node-zhg9b, kube-system/kustodian-kgmzw, kube-system/rancher-vsphere-cpi-cloud-controller-manager-s72z7, kube-system/rke2-ingress-nginx-controller-cvbz6, kube-system/vsphere-csi-node-fcbkn, kured-system/kured-pcg6l, longhorn-system/engine-image-ei-d474e07c-zjt9x, longhorn-system/longhorn-csi-plugin-vrh6r, longhorn-system/longhorn-manager-v5d76, metallb-system/load-balancer-metallb-speaker-s65sm
evicting pod metallb-system/load-balancer-metallb-controller-6f4f9544f6-6p7x7
evicting pod default/nginx-ingress-controller-nginx-ingress-84fbbf7747-rr24w
evicting pod longhorn-system/csi-resizer-68ccff94-c24bl
evicting pod longhorn-system/csi-snapshotter-6d7d679c98-8s7vl
evicting pod longhorn-system/instance-manager-e-05cce02d
evicting pod harbor-services/harbor-chartmuseum-cb6c9d47-bvfq6
evicting pod longhorn-system/csi-attacher-8b4cc9cf6-h2xl2
evicting pod longhorn-system/instance-manager-r-ab43ffa1
evicting pod harbor-services/harbor-core-544c9cf75b-flcns
evicting pod longhorn-system/csi-provisioner-59b7b8b7b8-q4mc7
evicting pod cert-manager/cert-manager-webhook-586d45d5ff-wtt6f
evicting pod longhorn-system/csi-attacher-8b4cc9cf6-4c8k7
error when evicting pods/"instance-manager-e-05cce02d" -n "longhorn-system" (will retry after 5s): Cannot evict pod as it would violate the pod's disruption budget.
error when evicting pods/"instance-manager-r-ab43ffa1" -n "longhorn-system" (will retry after 5s): Cannot evict pod as it would violate the pod's disruption budget.
I0609 07:08:46.309032    2487 request.go:665] Waited for 1.047973151s due to client-side throttling, not priority and fairness, request: GET:https://10.43.0.1:443/api/v1/namespaces/longhorn-system/pods/csi-attacher-8b4cc9cf6-h2xl2
evicting pod longhorn-system/instance-manager-e-05cce02d
error when evicting pods/"instance-manager-e-05cce02d" -n "longhorn-system" (will retry after 5s): Cannot evict pod as it would violate the pod's disruption budget.
evicting pod longhorn-system/instance-manager-r-ab43ffa1
error when evicting pods/"instance-manager-r-ab43ffa1" -n "longhorn-system" (will retry after 5s): Cannot evict pod as it would violate the pod's disruption budget.
... (last two lines repeated about 50 times)
```

**Comment by ckotzbauer** [comment]
Ok, so kured detected a sentinel-file, can you share the top of the logs (or your kured-config) too please?

**Comment by johnwc** [comment]
I updated the previous comment to include the top of the log. The odd thing though, I just connected to the server and it is showing uptime of 15 days. Something in kube on the node is keeping it from rebooting, it still has the status of cordoned.

**Comment by ckotzbauer** [comment]
Small summary of your info:
- There's a reboot notification from kured and you don't know why.
- The config says, the node can be rebooted at any time on any day!
- At `2022-06-09T07:08:38Z` kured detected the sentinel file and started draining the node
- The drain seems to be still in progress (the node is cordoned at that time), cause the instance-manager from longhorn is still present.
- The uptime of 15 days is because the node was not rebooted yet (as the drain did not finish)

So, as the drain seems not to finish automatically (wrong PDBs or sth. else why the pod cannot be evicted) you maybe have to kick the longhorn-pod manually. Then the reboot should be performed with an uncordon after that.

I don't see a misbehaviour from kured so far.

**Comment by johnwc** [comment]
There was a mix of info here, some not related to this issue, sorry for that. It not rebooting because of the failure to drain has nothing to do with this issue. At first I thought kured notified us that it was rebooting the node, when it didn't. But that was an issue of not being able to be drained. The issue is that there was never a sentinel file, so kured should not have rebooted it.

**Comment by johnwc** [comment]
I just figured out what was causing the issue. I went into the node that is stuck trying to drain, that kured tried to reboot. I noticed that there is a sentinel file along with `reboot-required.pkgs` file. It seems that ubuntu's automatic update service is enabled. It did an update in the middle of the night, creating the two files. We usually use a RHEL distro for our servers, but found it easier to install Rancher K8s on ubuntu cloud image. I completely forgot that ubuntu had a automatic update feature like Windows. We disabled that feature, in a production environment we need to schedule updates so they are planed and known about.
